lib/experiment/local/client.rb

```diff
@@ -61,15 +77,20 @@ def stop
 
     private
 
-    def parse_results(result, flag_keys)
+    def parse_results(result, flag_keys, user)
       variants = {}
+      assignments = {}
       result.each do |key, value|
-        next if value['isDefaultVariant'] || (flag_keys.empty? && flag_keys.include?(key))
+        included = flag_keys.empty? || flag_keys.include?(key)
+        if !value['isDefaultVariant'] && included
+          variant_key = value['variant']['key']
+          variant_payload = value['variant']['payload']
+          variants.store(key, Variant.new(variant_key, variant_payload))
+        end
 
-        variant_key = value['variant']['key']
-        variant_payload = value['variant']['payload']
-        variants.store(key, Variant.new(variant_key, variant_payload))
+        assignments[key] = value if included || value['type'] == 'mutual-exclusion-group' || value['type'] == 'holdout-group'
       end
+      @assignment_service&.track(Assignment.new(user, assignments))
```

Comment:
I'd prefer if we pull this track call into the evaluate function and use the assignments that return from this function call.

I.e. in evaluate()
```
assignments = parse_results(result, flag_keys)
@assignment_service&.track(Assignment.new(user, assignments))
assignments
```

Reply:
Here, it might be a bit tricky, since `parse_results` returns variants which is returned by `evaluate`, and the set of variants might be different from the set of assignments